src/v/coding-style.md

```diff
@@ -6,12 +6,50 @@ The document covers style and convention not enforced by clang-format.
 
 Header files have the `.h` extension, source files use the `.cc` extension.
 
+Public header files should go in an `includes` directory within the module they belong to. 
+For example, a module named `foo` should have public header files within `src/v/foo/include/foo`.
+
 Test files go in the `tests` directory of the module they belong to. That is,
 the tests for a feature named `foo` and located in `src/v/foo` will be located
 in `src/v/foo/tests`.
 
 Use `snake_case` naming convention by default for files and language identifiers.
 
+## Documentation
+
+### Modules
+
+Modules themselves should have a `README.md` file explaining the purpose of the module,
+it's API surface, and an architectural overview of that module (mentioning major classes),
+etc. This will aid people new to the module in understanding the lay of the land. When in doubt
+of if information should be placed in a class or in a README, prefer moving the documentation
+as close to the source as possible (inside the class documentation). We want to prioritize
+documentation not becoming stale - and also not cognitively overload the reader of the module
+README.
+
+### Classes
+
+Classes in public headers should be documented, plain structs that only hold data are not
```

Comment:
Why an exception for plain structs? They seem as worthy of class-level documentation as classes?